micrometer-core/src/test/java/io/micrometer/core/instrument/binder/jpa/HibernateQueryMetricsTest.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2019 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.jpa;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.MockClock;
+import io.micrometer.core.instrument.simple.SimpleConfig;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import org.hibernate.stat.QueryStatistics;
+import org.hibernate.stat.Statistics;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for {@link HibernateQueryMetrics}.
+ *
+ * @author Pawel Stepien
+ */
+class HibernateQueryMetricsTest {
+
+    private MeterRegistry registry;
+
+    @BeforeEach
+    void setup() {
+        registry = new SimpleMeterRegistry(SimpleConfig.DEFAULT, new MockClock());
+    }
+
+    private HibernateQueryMetrics.MetricsEventHandler createMetricsEventHandlerMock() {
+        HibernateQueryMetrics hibernateQueryMetricsMock = Mockito.mock(HibernateQueryMetrics.class);
```

Comment:
It feels like we're mocking too much when we are even mocking the class under test. I will try to rework the tests a bit to get something that feels closer to real usage. We could even make tests that use an in-memory database and normal Hibernate with that. This can be done post-merge, though.

Reply:
After looking at this more closely, I see we're essentially testing the `MetricsEventHandler` in isolation. That's fine for this. I'll open an issue to add tests with not-mocked Hibernate classes.